kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jValueConverterTest.kt

```diff
@@ -49,6 +54,167 @@ class Neo4jValueConverterTest {
         assertEquals(expected, result)
     }
 
+    @Test
+    fun `should convert tree with mixes types into map of neo4j values`() {
+
+        val utc = ZoneId.of("UTC")
+        val result = Neo4jValueConverter().convert(Struct(TEST_SCHEMA)) as Map<String, Value?>
+
+        val target = result["target"]
+        assertTrue{ target is FloatValue }
+        assertEquals(123.4, target?.asDouble())
+
+        val largeDecimal = result["largeDecimal"]
+        assertTrue{ largeDecimal is StringValue }
+        assertEquals(BigDecimal.valueOf(Double.MAX_VALUE).pow(2).toPlainString(), largeDecimal?.asString())
+
+        val byteArray = result["byteArray"]
+        assertTrue{ byteArray is BytesValue }
+        assertEquals("Foo".toByteArray().map { it }, byteArray?.asByteArray()?.map { it })
+
+        val int64 = result["int64"]
+        assertTrue{ int64 is IntegerValue }
+        assertEquals(Long.MAX_VALUE, int64?.asLong())
+
+        val int64Timestamp = result["int64Timestamp"]
+        assertTrue{ int64Timestamp is LocalDateTimeValue }
+        assertEquals(Date.from(Instant.ofEpochMilli(789)).toInstant().atZone(utc).toLocalDateTime(), int64Timestamp?.asLocalDateTime())
+
+        val int32 = result["int32"]
+        assertTrue{ int32 is IntegerValue }
+        assertEquals(123, int32?.asInt())
+
+        val int32Date = result["int32Date"]
+        assertTrue{ int32Date is DateValue }
+        assertEquals(Date.from(Instant.ofEpochMilli(456)).toInstant().atZone(utc).toLocalDate(), int32Date?.asLocalDate())
+
+        val int32Time = result["int32Time"]
+        assertTrue{ int32Time is LocalTimeValue }
+        assertEquals(Date.from(Instant.ofEpochMilli(123)).toInstant().atZone(utc).toLocalTime(), int32Time?.asLocalTime())
+
+    }
+
+    @Test
+    fun `should convert BigInteger into String value if is less then Long MIN_VALUE`() {
+
+        val number = BigInteger.valueOf(Long.MIN_VALUE).pow(2)
+        val result = Neo4jValueConverter().convert(getItemElement(number))
+        val item = result["item"]
+
+        assertTrue{ item is StringValue }
+        assertEquals(number.toString(), item?.asString())
+    }
+
+    @Test
+    fun `should convert BigInteger into String value if is greater then Long MAX_VALUE`() {
+
+        val number = BigInteger.valueOf(Long.MAX_VALUE).pow(2)
+        val result = Neo4jValueConverter().convert(getItemElement(number))
+        val item = result["item"]
+
+        assertTrue{ item is StringValue }
+        assertEquals(number.toString(), item?.asString())
+    }
+
+    @Test
+    fun `should convert BigInteger into Long neo4j value if is equal to Long MAX_VALUE`() {
```

Comment:
qual'è l'obiettivo di questo test?
Cosa succede se è minore di Long. MAX_VALUE?

Reply:
Sì, è più che altro un di più per far capire che fino a Long. MAX_VALUE compreso, il valore rimane un intero, mentre se viene superato diventa una stringa. Potrei anche toglierlo e mettere al test di sotto `should convert BigInteger into Long neo4j value if is less then OR EQUAL TO Long MAX_VALUE and greater then OR EQUAL TO Long MIN_VALUE`?